src/app/core/authentication/authentication.service.ts

```diff
@@ -1,142 +1,58 @@
-              variables: {
-                objects: [
-                  {
-                    email: authResult.idTokenPayload.email,
-                    name: authResult.idTokenPayload.name,
-                    username: authResult.idTokenPayload.nickname,
-                    profile_pic: authResult.idTokenPayload.picture
-                  }
-                ]
-              }
-            })
-            .subscribe(data => {
-              localStorage.setItem('commentsLikedByUser', JSON.stringify([]));
-              localStorage.setItem('user_profile_pic', data.data.insert_user.returning[0].profile_pic);
-              localStorage.setItem('username', data.data.insert_user.returning[0].username);
-              localStorage.setItem('userId', data.data.insert_user.returning[0].id);
-            });
-        } else {
-          this.apollo
-            .watchQuery<any>({
-              query: QUERY_USER_LIKES,
-              variables: {
-                userId: res.data.user[0].id
-              }
-            })
-            .valueChanges.subscribe((likes: any) => {
-              const likedComments: any = [];
-              const upvotedIdeas: any = [];
-              const likedProjects: any = [];
-              const likedReplies: any = [];
+  async logout() {
+    await this.afAuth.auth.signOut();
+    localStorage.clear();
+  }
 
-              likes.data.user[0].comment_likes.forEach((commentUserLikes: any) => {
-                likedComments.push(commentUserLikes.comment_id);
-              });
-              likes.data.user[0].ideaLikessByuserId.forEach((commentUserLikes: any) => {
-                upvotedIdeas.push(commentUserLikes.idea_id);
-              });
-              likes.data.user[0].projectLikessByuserId.forEach((commentUserLikes: any) => {
-                likedProjects.push(commentUserLikes.project_id);
-              });
-              likes.data.user[0].replyLikessByuserId.forEach((replyUserLikes: any) => {
-                likedReplies.push(replyUserLikes.reply_id);
-              });
+  private updateUserData(user: any) {
+    const userRef: AngularFirestoreDocument<any> = this.afs.doc(`users/${user.uid}`);
 
-              localStorage.setItem('ideaUpvotedByLoggedInUser', JSON.stringify(upvotedIdeas));
-              localStorage.setItem('commentsLikedByLoggedInUser', JSON.stringify(likedComments));
-              localStorage.setItem('projectsLikedByLoggedInUser', JSON.stringify(likedProjects));
-              localStorage.setItem('repliesLikedByLoggedInUser', JSON.stringify(likedReplies));
-            });
-          localStorage.setItem('user_profile_pic', res.data.user[0].profile_pic);
-          localStorage.setItem('username', res.data.user[0].username);
-          localStorage.setItem('userId', res.data.user[0].id);
-        }
-      });
-  }
+    const data = {
+      uid: user.uid,
+      email: user.email,
+      displayName: user.displayName,
+      photoURL: user.photoURL
```

Comment:
User roles needed imo

Reply:
All those are being taken care by hasura db. It's not needed here.